controlplane/src/core/composition/composer.ts

```diff
@@ -195,4 +195,51 @@ export class Composer {
       compositions: composedGraphs,
     };
   }
+
+  /**
+   * Same as compose, but without the subgraph that is being deleted
+   */
+  async composeWithDeletedSubGraph(subgraphLabels: Label[], subgraphName: string): Promise<CompositionResult> {
```

Comment:
This will duplicate too much logic. You could overload `composeFederatedGraph` with an optional excludeSubgraphs option.

Reply:
I can't re-use composeFederatedGraph since the checks compose based on graph labels.

Reply:
Improved code re-usage